go/service/assets/api_op_CreateAsset.go

```diff
@@ -52,35 +54,42 @@ func (c *Client) CreateAsset(createAssetInput CreateAssetInput) (*CreateAssetOut
 	createAssetApiInputJson, err := json.Marshal(createAssetApiInput)
 
 	if err != nil {
-		return nil, fmt.Errorf("unexpected error while marshaling create asset input: %v", err)
+		return nil, errors.FromError(err).FailedTo(internal.ACTION_JSON_MARSHAL)
 	}
 
-	req, err := http.NewRequest("POST", endpoint, bytes.NewBuffer(createAssetApiInputJson))
+	req, err := http.NewRequest(http.MethodPost, endpoint, bytes.NewBuffer(createAssetApiInputJson))
 	if err != nil {
-		return nil, fmt.Errorf("unexpected error while building HTTP request: %v", err)
+		return nil, errors.FromError(err).FailedTo(internal.ACTION_HTTP_BUILD_REQUEST)
 	}
 
 	res, err := c.HTTPClient.Do(req)
 	if err != nil {
-		return nil, fmt.Errorf("error while performing POST request to the endpoint %v: %v", endpoint, err)
+		return nil, errors.FromError(err).FailedTo(internal.ACTION_HTTP_DO_REQUEST)
 	}
 
 	defer res.Body.Close()
 	response, err := io.ReadAll(res.Body)
 	if err != nil {
-		return nil, fmt.Errorf("error while reading response body: %v", err)
+		return nil, errors.FromError(err).FailedTo(internal.ACTION_HTTP_READ_BYTES)
 	}
 
 	if res.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("error: got %d from %s %s endpoint . Full response : \n %s", res.StatusCode, res.Request.Method, endpoint, response)
+		// The assets API returns error in an array.
+		var v []internal.ConnectorApiError
+		err = json.Unmarshal(response, &v)
+		if err != nil {
+			return nil, errors.FromError(err).FailedTo(internal.ACTION_JSON_UNMARSHAL)
+		}
```

Comment:
I think we should always return the err message we received from the API, we can convert it to a string.

Reply:
> I think we should always return the err message we received from the API, we can convert it to a string.

The message we received from the API is returned (line 84).